### PR TITLE
feat: add configuration files and helloWorld function

### PR DESCRIPTION
**Pull request by carolestrella**
#### What is the purpose of this pull request?
Create the configuration files (nodemon, eslint, gitignore, jest, tsc and package.json) to the TypeScript template and test the helloworld function with the CI configuration.

#### What problem is this solving?
Be able to start a TypeScript project with all the configuration files set.

#### Types of changes
- [ ] Refactor (non-breaking change that only makes the code better)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

#### Chores checklist
- [ ] Update `CHANGELOG.md`

### CODE REVIEW

**Comment by vitordullens** [comment]
### Observation
Maybe the ideal would be to separate the test and lint part into separate jobs on CI.yml, so that you can have a deeper knowledge of what part of the code is wrong. 😄

**Comment by VerasThiago** [comment]
How about the branch name with the base prefix ? `fix/` / `feature/` / `refactor/` ... ?

**Comment by VerasThiago** [comment]
You need to set this repo to require at lease 1 review.

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+node_modules
+build
+build
```
Comment (VerasThiago) [inline .gitignore:3]: Build duplicated

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#CHANGE LOG
```
Comment (VerasThiago) [inline CHANGELOG.md:1]: This `CHANGELOG` should be empty (only with the default template), can you explain me why ?

File: jest.config.js [hunk 3]
```diff
@@ -0,0 +1,188 @@
+// For a detailed explanation regarding each configuration property, visit:
```
Comment (VerasThiago) [inline jest.config.js:1]: How about removing this comments?

File: templates/issue_template [hunk 4]
```diff
@@ -0,0 +1,17 @@
+### Subject of the issue
```
Comment (VerasThiago) [inline templates/issue_template:1]: This file should be inside `.git/` folder

File: templates/pull_request_template [hunk 5]
```diff
@@ -0,0 +1,19 @@
+#### What is the purpose of this pull request?
```
Comment (VerasThiago) [inline templates/pull_request_template:1]: This file should be inside `.github/` folder

File: src/index.ts [hunk 6]
```diff
@@ -0,0 +1,8 @@
+function helloworld (msg = 'Hello World!'): string {
```
Comment (VerasThiago) [inline src/index.ts:1]: The ideia in the `index` is to import the `helloWorld` function from `modules` folder so then you can call `helloWorld()`

File: src/index.ts [hunk 7]
```diff
@@ -0,0 +1,8 @@
...
+  console.log(msg)
+  return msg
+}
+
+helloworld()
+
+export default helloworld
```
Comment (VerasThiago) [inline src/index.ts:8]: Why you exported this function ? No one is using it.

Comment (VerasThiago) [inline src/index.ts:8]: Also take a look of the difference between `export default` and `export` and explain here, please.

Comment (carolestrella) [reply src/index.ts:8]: I needed to export it so that I could use it on tests.

File: src/index.ts (hunk 6 above)
Comment (carolestrella) [reply src/index.ts:1]: Ok, just fixed it.

File: CHANGELOG.md (hunk 2 above)
Comment (carolestrella) [reply CHANGELOG.md]: Because the project has not yet been released.

File: src/index.ts (hunk 7 above)
Comment (carolestrella) [reply src/index.ts]: Both can be used to export constants, functions, files and modules but `Export` is used to export multiple objects while `export default` is used for a single export.

**Pull request closed without merging**